JSQMessagesViewController/Views/JSQMessagesCollectionView.m

```diff
@@ -95,6 +100,18 @@ - (void)awakeFromNib
     [self jsq_configureCollectionView];
 }
 
+#pragma mark - Overrides
+
+- (void)layoutSubviews {
+    if (!CGSizeEqualToSize(previousSize, self.frame.size)) {
+        JSQMessagesCollectionViewFlowLayoutInvalidationContext *context = [JSQMessagesCollectionViewFlowLayoutInvalidationContext context];
+        context.invalidateFlowLayoutMessagesCache = YES;
+        [self.collectionViewLayout invalidateLayoutWithContext:context];
+        previousSize = self.frame.size;
+    }
+    [super layoutSubviews];
+}
```

Comment:
Yes I also was a bit afraid of getting an infinite loop or something like this by invalidating a layout in ```-layoutSubviews```. But no problems so far.

I was considering ```-setFrame:``` but it looks like it isn't a good idea. Take a look here: http://stackoverflow.com/questions/4000664/is-there-a-uiview-resize-event
, esp. this answer and comments: http://stackoverflow.com/a/23277135